### ISSUE TITLE
Improve reliability of simulated clicks

### ISSUE THREAD

**Issue by mjrusso** [mjrusso/scoot#28]
The current click simulation code is written as follows ([source](https://github.com/mjrusso/scoot/blob/8f7643969ee54600585f298ac7ad1791c5bd8ba1/Scoot/Mouse.swift#L39-L68)):

```swift
    func click(button: Mouse.Button) {
        // It should be possible to `pressDown()` and then `release()` (after a
        // short delay), however that approach doesn't work reliably: in some
        // applications (for example, when trying to click on a hyperlink in a
        // browser, or in an apps that render their user interface using web
        // technologies), mouse clicks don't actually register.
        //
        // For an unknown reason, posting two `leftMouseDown` events, followed
        // by a single `leftMouseUp`, *seems* to work consistently everywhere.
        // How's that for a hack?
        //
        // [N.B.: A previous implementation issued two clicks in a row (i.e.,
        // `pressDown()`, `release()`, `pressDown()`, `release()`), and that
        // worked better than one could reasonably expect. However, there were
        // rare cases where two clicks would actually be registered: for example,
        // in Safari, clicking the "close tab" button (the one embedded in the
        // tab itself) would result in two tabs being closed. Another example:
        // it wasn't possible to use Scoot to bring up it's menu bar: two clicks
        // would register, and it would immediately close.]
        //
        // [N.B.: This question [0] on Stack Overflow ("Simulating mouse clicks
        // on Mac OS X does not work for some applications") seems relevant, but
        // unfortunately none of the proposed strategies seem to work.]
        //
        // [0]: https://stackoverflow.com/q/2369806/15304124
        pressDown(button)
        pressDown(button)
        usleep(40000)
        release(button)
    }
```

I'd love to understand why this hack is necessary, and ideally modify the implementation to `pressDown` only a single time.

**Comment by mjrusso** [comment]
I have a hunch that the two `pressDown`s can cause problems in certain apps. For example. see #27; it isn't possible to drag-and-drop in the Finder window that pops up when you double click on a DMG file, and I wouldn't be surprised if it's because of the double `pressDown`.

**Comment by diegolhambi** [comment]
Maybe the click don't work because the target window is out of focus?

I was testing the Scoot and warpd to check why the warpd works and I suposse is because the warpd wont get the focus to himself.

**Comment by mjrusso** [comment]
@diegolhambi my apologies, I completely missed your reply. Thanks for this! Bringing Scoot to the background before left clicking makes a lot of sense. 

Just to clarify, when you say this change makes Firefox work reliably, is that also in conjunction with removing one of the  `pressDown` function calls in the `click` method? Or are you seeing some other general problem with Scoot in Firefox?

**Comment by diegolhambi** [comment]
Hey @mjrusso don't worry thanks for replying.

Yes is in conjunction with removing one of the `pressDown` and reducing the `usleep` to `10000`.
![image](https://user-images.githubusercontent.com/10106329/205641906-261c83c5-a2ea-4da6-8977-ea246669ec6f.png)

I'm using with these modifications for a whole month and is working flawlessly but I only use the grid and free style for scrolling and precise click.

**Comment by mjrusso** [comment]
Fixes shipped as part of [v1.2](https://github.com/mjrusso/scoot/releases/tag/v1.2) :)
